Samples/NetworkMediaStreamer/Readme.md

```diff
@@ -0,0 +1,298 @@
+---
+### Return  
+>HRESULT 
+: S_OK if succeeded. HRESULT error if failed. 
+---
+
+## Authentication provider
+
+The authentication provider implements the following interfaces 
+1. IRTSPAuthProvider: This interface is used by the RTSP server to authenticate incoming client connections
+2. IRTSPAuthProviderCredStore: This interface is used by the administrative module of the app to add and remove credentials etc.  
+A base/sample implementation of AuthProvider is included in the source. An instance to the authentication provider can be created using -
+---
+```
+ HRESULT GetAuthProviderInstance(
+    AuthType authType,
+    LPCWSTR pResourceName,
+    IRTSPAuthProvider** ppRTSPAuthProvider);
+``` 
+### Arguments  
+|  |  |
+| ----------- | ----------- |
+| authType | Input array of MFMediaType objects which describe each stream of the sink |
+| pResourceName | Input resource name to which the credentials are bound |
+| ppRTSPAuthProvider | Output pointer to the IRTSPAuthProvider interface to the auth provider instance |
+---
+### Return  
+>HRESULT 
+: S_OK if succeeded. HRESULT error if failed. 
+---
+
+## Interface definitions
+---
+### IRTSPAuthProvider
+```
+IRTSPAuthProvider : public ::IUnknown
+{
+    virtual STDMETHODIMP GetNewAuthSessionMessage(HSTRING* pAuthSessionMessage) = 0;
+    virtual STDMETHODIMP Authorize(LPCWSTR pAuthResp, LPCWSTR pAuthSesMsg, LPCWSTR pMethod) = 0;
+};
+```
+`IRTSPAuthProvider::GetNewAuthSessionMessage(HSTRING* pAuthSessionMessage)`  
+Generates nonce and Gets new authentication request message from Auth provider
+
+| |  |  |
+| ----------- | ----------- | --------- |
+| pAuthSessionMessage | Output pointer to the Authentication request/description message to be sent to client as per [RFC7616](https://tools.ietf.org/html/rfc7616#section-3.7). | e.g. `WWW-Authenticate: Digest realm="BeyondTheWall", nonce="6fc93a0604338c68070b75e49745baeab46c507155040000", algorithm=MD5, charset="UTF-8", stale=FALSE` |
+| return value | S_OK if succeeded. HRESULT error if failed. | 
+| 
+
+`IRTSPAuthProvider::Authorize(LPCWSTR pAuthResp, LPCWSTR pAuthSesMsg, LPCWSTR pMethod)`  
+Verifies authorization response received from the client  
+||||
+| ----------- | ----------- | -------- |
+| pAuthResp | Input pointer to Authorization response received from client| e.g. `Authorization: Digest username="user", realm="BeyondTheWall", nonce="8e03237a99bc26d6914a562fc98b08593fec11a055040000", uri="rtsp://127.0.0.1:8554", response="742dd5a18cddb799a861461ae72570b2"`|
+| pAuthSesMsg | Input pointer to the authentication request that was sent to client | e.g. `WWW-Authenticate: Digest realm="BeyondTheWall", nonce="6fc93a0604338c68070b75e49745baeab46c507155040000", algorithm=MD5, charset="UTF-8", stale=FALSE` |
+| pMethod | Input pointer to the name of the request method to be authenticated | e.g. `PLAY`| 
+| return value | S_OK if succeeded. HRESULT error if failed. | `HRESULT_FROM_WIN32(ERROR_INVALID_PASSWORD)  if authentication fails.`
+

```

Comment:
For some method prototypes, we use this earlier in documentation. Consistency?

### Return  
>HRESULT 
: S_OK if succeeded. HRESULT error if failed.